### DISCUSSION TITLE
서비스 코드의 메서드 배치에 관한 의견

### DISCUSSION THREAD

**Discussion by JJimini** [BDD-CLUB/01-doo-re-back#193] [논의 사항]
현재 DOORE 서비스에서는 **서비스 코드의 메서드 위치**에 대한 규칙은 따로 없습니다.
하지만 자유롭게 작성하니, 코드 수정을 하면서 메서드가 헷갈리는 일이 많았습니다. (거의 `public`이랑 `private`으로 구분하는 정도..)

따라서 저는 메서드 위치를 통일하는 방안에 대해 어떤지 의견을 묻고 싶습니다.

**1. `Controller`에 직접 연결되는 메서드(`public`으로 적혀있는 것들)를 `Controller`와 같은 순서대로 작성하고, 밑에 메서드에서 호출하는 메서드(`private`로 적혀있는 것들)를 `Controller `순서와 연관하여 작성**
2. `Controller`에 직접 연결되는 메서드를 순서대로 적되, 그 메서드에서 호출하는 메서드를 연관 메서드 밑에 작성

2번 방법은 바로 밑에 연관되는 메서드가 있어 코드 읽기가 편하다는 장점이 있지만, 그 `private` 메서드를 다른 메서드에서도 호출하는 경우 이 장점은 없어진다고 생각합니다.
또한, 대부분의 개발자들이 바로가기(단축키, 링크)를 통해 메서드를 이동해다니기 때문에(제 생각일수도...), 2번 방법의 장점이 잘 들어나지 않는 다는 점도 있습니다.

1번 방법을 사용한다면 서비스 코드를 보았을 때, 핵심이 되는 메서드를 위에서부터 한 눈에 알아볼 수 있다는 장점이 있습니다!
추가적으로 `class`파일의 이름에 맞는 메서드들이 위쪽으로, 아닌 것들이 아래쪽으로 배치되는 것이 구분하기 편할 것 같습니다!
예를들어, `curriculumService`의 경우 `Curriculum `관련이 위쪽으로 배치, `garden`관련은 아래쪽으로 배치

해당 사안에 대해 의견이 궁금합니다!

**Comment by lcqff** [comment]
답변 늦어서 죄송합니다! 확인을 못했네요😥

개인적으로 2번 방식으로 코드를 작성하고 있긴 했습니다. 코드 리뷰를 할 때 브랜치를 이동해서 IDE로 확인하는게 귀찮았기 때문에 PR 변경사항을 통해 확인하는 편인데 이때 2번 방식이 제일 코드를 읽기 쉬웠다는 느낌입니다.
사실 저도 단축키를 통해 메서드를 확인하는 편이라 Service 메서드를 처음부터 읽는 일이 없었어서 어느 쪽으로 배치를 해도 크게 상관없다는 의견입니다😅

단 지금까지 작성한 코드의 배치도 수정한다면 2번이 더 낫지 않나, 이미 있는 코드는 내버려두고 지금부터 해당 규칙을 적용해 코드를 작성하자는 의견이라면 1번도 상관없지 않나 생각합니다. 지금은 참여하지 않는 다른 분들도 2번 방식을 사용해 코드를 작성하셨던거 같아서...(확인해본건 아니고 2번 방식으로 코드를 짰을때 리뷰하기 편하다고 칭찬을 들었던 기억이 있어서 그렇게 짐작하고 있습니다) 1번 방식으로 이미 작성된 코드를 수정하려면 변경사항이 많을 거 같아서요!

**Comment by JJimini** [comment]
변경사항이 많으면 혼란을 초래할 수도 있겠네요. 그렇다면 있는 코드는 내버려두고 지금부터 1번 사항을 권고하는 것은 어떨까요?? 어짜피 원래 코드는 그대로 있으니 규칙이라고 하기엔 좀 애매하고 권고 사항 정도로요!

**Comment by lcqff** [comment]
좋습니다~~~! 👍
